servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1476,6 +1520,183 @@ public void setState(long fileId, SetStateOptions options)
     }
   }
 
+  /**
+   * Schedules a file for async persistence.
+   *
+   * @param the id of the file for persistence
+   * @throws FileDoesNotExistException when the file does not exist
+   */
+  public synchronized void scheduleAsyncPersistence(long fileId) throws FileDoesNotExistException {
+    // find the worker
+    long workerId = getWorkerStoringFile(fileId);
+
+    // update the state
+    synchronized (mInodeTree) {
+      Inode inode = mInodeTree.getInodeById(fileId);
+      inode.setPersistenceState(FilePersistenceState.SCHEDULED);
+    }
+
+    if (!mWorkerToAsyncPersistFile.containsKey(workerId)) {
+      mWorkerToAsyncPersistFile.put(workerId, Sets.<Long>newHashSet());
+    }
+    mWorkerToAsyncPersistFile.get(workerId).add(fileId);
+  }
+
+  /**
+   * Gets the worker where the given file is stored.
```

Comment:
What is the exact semantics of this method? Is it any worker that stores any part of the file? or is it a worker that stores all of the file?

What will happen if the file is distributed across multiple workers?